v2/observer/workers_test.go

```diff
@@ -0,0 +1,395 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	myk8s "github.com/brigadecore/brigade/v2/internal/kubernetes"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestSyncWorkerPods(t *testing.T) {
+	const testNamespace = "foo"
+	const testPodName = "bar"
+
+	var syncWorkerPodFnCallCount int
+	var syncDeletedPodFnCalled bool
+	mu := &sync.Mutex{}
+
+	kubeClient := fake.NewSimpleClientset()
+
+	observer := &observer{
+		kubeClient: kubeClient,
+		syncWorkerPodFn: func(_ interface{}) {
+			mu.Lock()
+			defer mu.Unlock()
+			syncWorkerPodFnCallCount++
+		},
+		syncDeletedPodFn: func(_ interface{}) {
+			mu.Lock()
+			defer mu.Unlock()
+			syncDeletedPodFnCalled = true
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second)
+	defer cancel()
+
+	go observer.syncWorkerPods(ctx)
+
+	// The informer needs a little time to get going. If we don't put a little
+	// delay here, we'll be adding, updating, and deleting pods before the
+	// informer gets cranking.
```

Comment:
This shouldn't be an issue in actual deployments of Brigade, right?

Reply:
Correct